ccm_web/server/src/api/dtos/api.section.enrollment.dto.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ArrayMaxSize,
+  ArrayMinSize,
+  IsArray,
+  IsIn,
+  IsInt,
+  IsNotEmpty,
+  MaxLength,
+  ValidateNested
+} from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+import { Type } from 'class-transformer'
+import { UserEnrollmentType } from '../../canvas/canvas.interfaces'
+
+export class SectionEnrollmentDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  @MaxLength(255)
+  loginId: string
+
+  @ApiProperty({ enum: UserEnrollmentType })
+  @IsNotEmpty()
+  @IsIn(Object.values(UserEnrollmentType))
+  type: UserEnrollmentType
```

Comment:
Some of these properties are repeated from other DTOs. There appear to be ways to build variations on other DTOs using `mapped-types` (https://docs.nestjs.com/openapi/mapped-types#mapped-types). I think I may leave this the way it is for now, as I think we'll need to reexamine the set of DTOs once PR #247 is completed.

Reply:
Yeah... the mapped-types seems to keep with reusable and organized but not for this phase of the project. you can create an issue for this and put in the icebox

Reply:
See https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/313